azurerm/resource_arm_powerbidedicated_capacity_test.go

```diff
@@ -0,0 +1,236 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPowerBIDedicatedCapacityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPowerBIDedicatedCapacity_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPowerBIDedicatedCapacityExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "sku", "A1"),
+					resource.TestCheckResourceAttr(resourceName, "administrators.#", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAzureRMPowerBIDedicatedCapacity_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPowerBIDedicatedCapacityExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "sku", "A2"),
+					resource.TestCheckResourceAttr(resourceName, "administrators.#", "2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMPowerBIDedicatedCapacityExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Capacity not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).PowerBIDedicated.CapacityClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		if resp, err := client.GetDetails(ctx, resourceGroup, name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Capacity (Capacity Name %q / Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on PowerBIDedicated.CapacityClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPowerBIDedicatedCapacityDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).PowerBIDedicated.CapacityClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_powerbidedicated_capacity" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.GetDetails(ctx, resourceGroup, name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on CapacityClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPowerBIDedicatedCapacity_basic(rInt int, location string) string {
+	template := testAccAzureRMPowerBIDedicatedCapacity_template(rInt, location)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_powerbidedicated_capacity" "test" {
+  name                = "acctestpowerbidedicatedcapacity%d"
```

Comment:
If you feel this name is too long, I can rename it to "acctestpowerbicapacity". However, I cannot rename as what you suggested since this resource is to create capacity which is the feature of the power bi dedicated.